### ISSUE TITLE
IndexOutOfBoundsException is thrown when a class contains a method that returns anonymous class instances

### ISSUE THREAD

**Issue by ChristianSchwarz** [TNG/junit-dataprovider#46]
The following code snipped throws : ```java.lang.IndexOutOfBoundsException: No group 2````
Version 1.9.0
```
@RunWith( DataProviderRunner.class )
public class TestCase {
 
@Test
@UseDataProvider( "dataProvider" )
public void test( Object objects ) {
 
}
 
@DataProvider
public static Object[][] dataProvider() {
return new Object[][] { { m1() } };
}
 //if you comment this methos out it will run
private static Object m2() {
return new Object() {
};
}
 
private static Object m1() {
return new Object() {
};
}
 
} 
```

Stacktrace:

```
java.lang.IndexOutOfBoundsException: No group 2
	at java.util.regex.Matcher.group(Matcher.java:470)
	at java.util.regex.Matcher.appendReplacement(Matcher.java:737)
	at com.tngtech.java.junit.dataprovider.internal.placeholder.BasePlaceholder.process(BasePlaceholder.java:61)
	at com.tngtech.java.junit.dataprovider.DataProviderFrameworkMethod.getName(DataProviderFrameworkMethod.java:70)
	at org.junit.runners.BlockJUnit4ClassRunner.testName(BlockJUnit4ClassRunner.java:203)
	at org.junit.runners.BlockJUnit4ClassRunner.describeChild(BlockJUnit4ClassRunner.java:76)
	at org.junit.runners.BlockJUnit4ClassRunner.describeChild(BlockJUnit4ClassRunner.java:50)
	at org.junit.runners.ParentRunner.getDescription(ParentRunner.java:298)
	at org.junit.runner.Runner.testCount(Runner.java:41)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestClassReference.countTestCases(JUnit4TestClassReference.java:30)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.countTests(RemoteTestRunner.java:487)
```

**Comment by janschaefer** [comment]
You example works for me with version 1.9.3 using Java 1.6

**Comment by aaschmid** [comment]
Hi @ChristianSchwarz, 
please do not use version 1.9.0 because of two bugs (see also https://github.com/TNG/junit-dataprovider/releases/tag/v1.9.0). Sorry for the inconvience, but with v1.9.0 and above this should be fixed! But I suggest using 1.9.4 which contains several fixes.
Best regards,
Andreas

**Comment by ChristianSchwarz** [comment]
Thanks, i can confirm that it works with 1.9.4!
